shop_basket/views.py

```diff
@@ -1,73 +1,84 @@
 from django.shortcuts import render, redirect
-from django.views.generic import TemplateView
+from django.views.generic import TemplateView, FormView
 from django.contrib.auth.decorators import login_required
+from django.contrib.auth.mixins import LoginRequiredMixin
 from django.contrib import messages
+from django.db.models import Max, Avg, Sum, Q, F, CharField
+from django.db.models.functions import Length
+
+from typing import List, Dict, Any
```

Comment:
Imports ordering can be improved.

**Example**
`
Built-in modules

3-rd party modules

Project-specific modules
`

https://www.python.org/dev/peps/pep-0008/#imports